sherpa/plot/__init__.py

```diff
@@ -2595,9 +2697,20 @@ def calc(self, fit, par):
         plot, prepare
         """
 
-        if type(fit.stat) in (LeastSq,):
+        if isinstance(fit.stat, LeastSq):
```

Comment:
It is a **very** long-standing complaint I have about our code base that it is littered with checks like this rather than having more-principled ways of determining whether it can be used in this way (and it is mainly a complaint about the stats and optimization method class design). You've made me go and make an issue about it: #1966 

I don't see me working on this anytime soon, but still.